transports/tls/src/lib.rs

```diff
@@ -79,3 +80,23 @@ pub fn make_server_config(
 
     Ok(crypto)
 }
+
+/// Create a TLS server configuration for libp2p.
+pub fn make_server_config_with_cert(
+    certificate: Certificate,
+    private_key: PrivateKey,
+    alpn: Vec<Vec<u8>>
+) -> Result<rustls::ServerConfig, certificate::GenError> {
```

Comment:
We need to find a way to make generation of these certificates deterministic.

Reply:
See the corresponding go-libp2p pull request https://github.com/libp2p/go-libp2p/pull/1833.